### PR TITLE
OOP school library: add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by KingsleyIbe**
### In this milestone, I implemented the following changes according to the project requirement:

- Applied classes and objects in Ruby.
- Applied encapsulation and inheritance with Ruby.

### CODE REVIEW

File: person_class_obj.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class Person
+  def initialize(age, parent_permission: true, name: 'Unknown')
```
Comment (MahmoudBakr23) [inline person_class_obj.rb:5]: - [ ] Great job so far implementing the `Person` Class. However, I think you might have missed out to specify that the `:name` and `:age` should be the only instances that can be accessed and modified using the `attr_accessor` , and the `:id` instance should be only accessed and not modified using the `attr_reader`.

_I hope this makes sense to you_ 😄

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @KingsleyIbe 😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Good README, PR headline, and summary
- Linters are correct
- Great implementation and following the project guidelines so far

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Kindly remember to tag @MahmoudBakr23 in your comment!**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person_class_obj.rb (hunk 1 above)
Comment (KingsleyIbe) [reply person_class_obj.rb:5]: @MahmoudBakr23 

Thank you for reviewing my work, be sure suggested changes will be implemented.

**Review by dasileker — APPROVED** [review]
### **Hi @KingsleyIbe,** 

### **Congratulations 👏👏**

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚢 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by KingsleyIbe** [comment]
@dasileker 

Thanks for your prompt review and for approving my project.

**Pull request merged**